.github/workflows/integrate.yaml

```
@@ -17,25 +17,30 @@ jobs:
           
   lint:
     name: Lint
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-24.04
 
     steps:
     - name: Check out code
       uses: actions/checkout@v4
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v5.3.0
+      with:
+        python-version: 3.8
     - name: Install dependencies
-      run: |
-        set -eux
-        sudo apt update
-        sudo apt install tox
+      run: pip install tox
     - name: Lint code
       run: tox -vve lint
 
   unit:
     name: Unit Tests
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-24.04
     steps:
       - name: Check out code
         uses: actions/checkout@v4
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v5.3.0
```

nitpick: Let's use version `v5.4.0` which has been released as of Jan 28: https://github.com/actions/setup-python/releases/tag/v5.4.0
```suggestion
        uses: actions/setup-python@v5.4.0
```